chromium_src/chrome/browser/component_updater/widevine_cdm_component_installer.h

```diff
@@ -0,0 +1,36 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_CHROMIUM_SRC_CHROME_BROWSER_COMPONENT_UPDATER_WIDEVINE_CDM_COMPONENT_INSTALLER_H_
+#define BRAVE_CHROMIUM_SRC_CHROME_BROWSER_COMPONENT_UPDATER_WIDEVINE_CDM_COMPONENT_INSTALLER_H_
+
+#include "brave/components/widevine/static_buildflags.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+
+#ifdef RegisterWidevineCdmComponent
+#pragma push_macro("RegisterWidevineCdmComponent")
+#undef RegisterWidevineCdmComponent
+#define RegisterWidevineCdmComponent_WasDefined
+#endif
```

Comment:
what's the purpose of this block? It shouldn't be necessary.

Reply:
> what's the purpose of this block? It shouldn't be necessary.

Our `registration.cc` has `#define RegisterWidevineCdmComponent`. Then it includes upstream's `registration.cc`. This includes our `widevine_cdm_component_installer.h`. In this case, our `widevine_cdm_component_installer.h` encounters a situation where `RegisterWidevineCdmComponent` is still defined.

Reply:
> > what's the purpose of this block? It shouldn't be necessary.
> 
> Our `registration.cc` has `#define RegisterWidevineCdmComponent`. Then it includes upstream's `registration.cc`. This includes our `widevine_cdm_component_installer.h`. In this case, our `widevine_cdm_component_installer.h` encounters a situation where `RegisterWidevineCdmComponent` is still defined.

the right thing here is to include this file into our `registration.cc` before upstream does it. This way you won't get defines clash.

this is a pretty frequent issue that we solve by manually including headers that may conflict with defines in `chromium_src` files.

Reply:
Got it. Resolved in 8c7fd1e72078f29151280aef80ea95d8e6ee0f4e.